ACPs/224-dynamic-gas-limit-in-subnet-evm/README.md

```diff
@@ -0,0 +1,233 @@
+    "maxCapacityFactor": ...,
+    "timeToDouble": ...,
+    "targetBlockRate": ...,
+    "minBlockGasCost": ...,
+    "maxBlockGasCost": ...,
+    "blockGasCostStep": ...
+  }
+}
+
+```
+
+### New ACP176 Fee Manager Precompile
+
+A new fee manager precompile will be required to dynamically changing the parameters. The precompile will be similar to the existing fee manager precompile. The solidity interface will be as follows:
+
+```solidity
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+import "./IAllowList.sol";
+
+interface IACP176FeeManager is IAllowList {
+  struct ACP176FeeConfig {
+    uint256 targetGas;
+    uint256 minGasPrice;
+    uint256 maxCapacityFactor;
+    uint256 timeToDouble;
+    uint256 targetBlockRate;
+    uint256 minBlockGasCost;
+    uint256 maxBlockGasCost;
+    uint256 blockGasCostStep;
+  }
+  event FeeConfigChanged(address indexed sender, ACP176FeeConfig oldFeeConfig, ACP176FeeConfig newFeeConfig);
+
+  // Set fee config fields to contract storage
+  function setFeeConfig(
+    uint256 targetGas;
+    uint256 minGasPrice;
+    uint256 maxCapacityFactor;
+    uint256 timeToDouble;
+    uint256 targetBlockRate;
+    uint256 minBlockGasCost;
+    uint256 maxBlockGasCost;
+    uint256 blockGasCostStep;
+  ) external;
+
+  // Get fee config from the contract storage
+  function getFeeConfig()
+    external
+    view
+    returns (
+      uint256 targetGas;
+      uint256 minGasPrice;
+      uint256 maxCapacityFactor;
+      uint256 timeToDouble;
+      uint256 targetBlockRate;
+      uint256 minBlockGasCost;
+      uint256 maxBlockGasCost;
+      uint256 blockGasCostStep;
+    );
+
+  // Get the last block number changed the fee config from the contract storage
+  function getFeeConfigLastChangedAt() external view returns (uint256 blockNumber);
+}
+```
+
+The precompile will have a `targetGas` parameter that is used to set the target gas consumed per second for ACP-176. This will override the `gas-target` in node config. If the new fee manager precompile is not activated, validators can keep using `gas-target` in node config to set the target gas consumed per second (as in C-Chain).
+
+```mermaid
+flowchart TD
+    A[ACP-224 activated] --> B{Is ACP176FeeManager precompile active?}
+
+    B -- Yes --> C[Use targetGas from precompile storage]
+
+    B -- No --> D{Is gas-target set in node config?}
+    D -- Yes --> E[Use gas-target from node config]
+
+    D -- No --> F{Is targetGas set in chainConfig.acp176FeeConfig?}
+    F -- Yes --> G[Use targetGas from chain config]
+
+    F -- No --> H[Use default targetGas e.g. 2 million]
+```
+
+## Backwards Compatibility
+
+ACP-224 will require a network update in order to activate the new fee mechanism. Another activation will also be required to activate the new fee manager precompile. The activation of precompile should never occur before the activation of ACP-224 (the fee mechanism) since the precompile depends on ACP-224’s fee update logic to function correctly.
```

Comment:
Could these activations be scheduled one immediately (or one second) after each other? Or is there an edge case if there is no blocks produced in between their activations?

Reply:
Precompile activations itself is pretty much harmless, because we always use the fee config from the parent state. the only edge-case might be the deactivation of old precompile in the sense that how it should be done. i.e do we want to leave it to the configuratio, like expecting the configuration to disable old precompile first before activating new one. or else we can handle that in the activation of acp176 (force to disable the precompile). 

I added it as a open question.